src/controllers/utils/searchWordUsingIgbo.ts

```diff
@@ -8,11 +8,12 @@ import { getCachedWords, setCachedWords } from '../../APIs/RedisAPI';
 import { handleWordFlags } from '../../APIs/FlagsAPI';
 import Version from '../../shared/constants/Version';
 import { SearchRegExp } from '../../shared/utils/createRegExp';
-import WordClassEnum from '../../shared/constants/WordClassEnum';
+import { Filters } from '../types';
+import { Keyword } from './types';
 
 type IgboSearch = {
   redisClient: RedisClientType | undefined;
-  keywords: { text: string; wordClass: WordClassEnum[]; regex: Pick<SearchRegExp, 'wordReg'> | SearchRegExp }[];
+  keywords: Partial<Keyword>[];
```

Comment:
why are using `Partial` on `Keyword`? using `Partial` in this case will make other fields like `text` and `wordClass` optional, which is something we want to avoid doing since we want to require that when we're dealing with `Keyword` we can expect `text` and `wordClass` to exist - that's what the original type that you replaced was defining

Reply:
The initial definition of keyword I had was the reason for this Partial usage. I have made update.